src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,53 +1,179 @@
-import { useState } from 'react';
+import { useState, FormEvent, useEffect } from 'react';
 import { TextField } from '../TextField';
+import { Movie, IsMovie, SmallerMovieGroup } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
+interface Props {
+  newMovie: Movie,
+  touchedMovies: IsMovie,
+  setNewMovie: (newValue: Movie) => void,
+  setTouchedMovies: (newValue: IsMovie) => void,
+  valueDelete: () => void,
+  addOn: () => void,
+  isButtonDisabled: boolean,
+}
+
+export const NewMovie: React.FC<Props> = ({
+  newMovie,
+  touchedMovies,
+  setTouchedMovies,
+  addOn,
+  setNewMovie,
+  valueDelete,
+  isButtonDisabled,
+}) => {
   const [count] = useState(0);
+  const [hasError, setHasError] = useState(false);
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = newMovie;
+
+  type MovieKey = keyof SmallerMovieGroup;
+
+  type TouchedKeys = {
+    [key in MovieKey]: boolean;
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+  };
+
+  const addMovieComponent = (event: string, secondValue: string) => {
+    const key = secondValue;
+    const updatedObjectTitle = { ...newMovie, [key]: event };
+
+    setNewMovie(updatedObjectTitle);
+  };
+
+  const addTouchedComponent = (
+    event: boolean,
+    componentName: string,
+    isRequired: boolean,
+  ) => {
+    const compKey = componentName;
+    const updatedObjectDescription = { ...touchedMovies, [compKey]: event };
+    const zeroValue = newMovie[compKey as keyof IsMovie];
+
+    setTouchedMovies(updatedObjectDescription);
+
+    if (!zeroValue.length
+      && isRequired) {
+      setHasError(true);
+    }
+  };
+
+  useEffect(() => {
+    const properties: MovieKey[]
+     = ['title', 'imgUrl', 'imdbUrl', 'imdbId'];
+    const update: TouchedKeys = { ...touchedMovies };
+
+    properties.forEach((property) => {
+      if (newMovie[property].length > 0) {
+        update[property] = false;
+      }
+    });
+
+    setTouchedMovies(update);
+  }, [newMovie]);
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
+      {}
+
       <TextField
```

Comment:
```suggestion
      <h2 className="title">Add a movie</h2>

      <TextField
```
What's the purpose of curly brackets here?